### ISSUE TITLE
Oper data:  Not able set leaf containing SR_UINT32_T/SR_INT32_T 

### ISSUE THREAD

**Issue by bhatkj** [sysrepo/sysrepo#661]
I am facing the issue wherein if i try to set leaf of oper  data having data type as  SR_UINT32_T or SR_INT32_T via *sr_val_set_str_data* api then it is getting stuck which in-turn making data provider in dead lock state. 
However against the same leaf if i change to STRING or ENUM then it is working. 
I couldn't find another api other than **sr_val_set_str_data** to set value. 
So trying to set in following way 
```
        sr_val_set_str_data(&v[1], SR_UINT32_T, "100");
```
 
Do we have any other api w.r.to oper data?

**Comment by rastislavszabo** [comment]
Hi,
`sr_val_set_str_data` is supposed to be used for string-based data types, which integer is not. Therefore you are getting (or at least you should) `SR_ERR_INVAL_ARG` return value from `sr_val_set_str_data`.

setting of non-string data should be direct:
```
value.data.uint32_val = 100;
```

**Comment by bhatkj** [comment]
Hi, 

Thank you.  
Ok. But just wondering on SR_ERR_INVAL_ARG , why it should get stuck. Ideally it should report error so that further operation can be performed.

**Comment by rastislavszabo** [comment]
Yes, it should not stuck. If really stucks *inside* of the `sr_val_set_str_data` than that is some weird bug.

**Comment by bhatkj** [comment]
Just now with  v[2].data.uint32_val = 100;  But the behavior still same.

**Comment by rastislavszabo** [comment]
Than it seems that the code was not stuck inside of the `sr_val_set_str_data`, but somewhere else. You would need to share your code and possibly also traces, otherwise we can not advise you.

**Comment by bhatkj** [comment]
Hi,
 Further digging into this problem. I got the root cause. Basically callback/dataprovider gets stuck if we dont set the data.type .  

For your reference i am attaching the **oper_data_example.c** with my edits by extending support for **in-octets** . Now if you compile the code problem can be seen. And if you uncomment the code which sets type then problem goes away. 
So currently it is mandate to type other application will get stuck for ever without any trace. 

Basically in my case it needs to be as follows which resolves issue.  
```
    v[2].data.uint32_val = 100;
        v[2].type = SR_UINT32_T;
```


One suggestion, it would be helpful if we oper_data_example.c which exists in the repository includes the reference for data type other than string type (i.e. int,unit )

**Comment by rastislavszabo** [comment]
Thanks for the analysis, we will fix the issue - processing should not stuck by providing invalid data.
